entc/gen/graph.go

```diff
@@ -216,6 +216,26 @@ func (g *Graph) addNode(schema *load.Schema) {
 	g.Nodes = append(g.Nodes, t)
 }
 
+// M2MModels returns a unique list of all
+// M2M models in the graph
+func (g *Graph) M2MModels() []string {
+	var models []string
+	visited := map[string]struct{}{}
+	for _, n := range g.Nodes {
+		for _, e := range n.Edges {
+			if e.M2M() {
+				model := e.M2MModel()
+				if _, ok := visited[model]; ok {
+					continue
+				}
+				visited[model] = struct{}{}
+				models = append(models, model)
+			}
+		}
+	}
+	return models
+}
```

Comment:
Let's remove this functionality from here and keep it in the template as it has only one usage.

Reply:
@a8m by "keep it in the template", do you mean have this function be completely defined in the .tmpl files? I don't think Go's templates allow you to set values such as an empty struct to a map to ensure uniqueness. 

Or do you mean to just move it to another file or package?